scheduler/src/cook/cors.clj

```diff
@@ -7,9 +7,11 @@
 (defn same-origin?
   "Returns true if the request is from the same origin as the provided origin header"
   [{:keys [headers scheme]}]
-  (let [{:strs [host origin]} headers]
+  (let [{:strs [host origin x-forwarded-proto]} headers
+        scheme (or x-forwarded-proto
+                   (when scheme (name scheme)))]
```

Comment:
Do we need the `when scheme` check? Is there a scenario where both `x-forwarded-proto` and `scheme` are `nil`?

Reply:
I don't _think_ scheme would ever be nil, but `name` throws an NPE if the argument is null, so I wanted to be defensive.